src/js/background.js

```diff
@@ -176,33 +180,59 @@ function RealDebrid(warningPercentage, warningDays, splittingSize, torrentHost)
     this.selectionHandler = function(selection) {
         var urls = selection.split(" ");
         var regex = new RegExp(/((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=\+\$,\w]+@)?[A-Za-z0-9.-]+|(?:www.|[-;:&=\+\$,\w]+@)[A-Za-z0-9.-]+)((?:\/[\+~%\/.\w-_]*)?\??(?:[-\+=&;%@.\w_]*)#?(?:[\w]*))?)/);
+        var invalid;
         $.each(urls, function(index, url) {
             if (url.match(regex)) {
                 that.urlHandler(url);
+            } else {
+                invalid = 1;
             }
         });
+        if (invalid) {
+          nf.error("Selected text doesn't seem to contain valid download URL(s)");
+        }
     };
 
     this.urlHandler = function(url) {
+        nf.progress(++nf.progressNotificationId,"Downloading...","Getting URL",0);
+        var regex = new RegExp(/(\/folder\/)/ig);
         if (url.lastIndexOf('magnet:', 0) === 0) {
-            that.handleMagnet(url);
-        } else {
-            that.unrestrict(url, function(result) {
-                if (result.download) {
-                    that.download(result.download);
-                } else if (result.error == 1) {
-                    nf.error("Please make sure you have your API token set. Click here to go to real-debrid.com", function() {
-                        chrome.tabs.create({
-                            url: URL_TOKEN
-                        }, function() {});
+            that.handleMagnet(url, function(result) {
+                if (result.uri) {
+                    chrome.tabs.create({
+                        url: "https://real-debrid.com/torrents"
                     });
                 } else {
-                    nf.basic(result.message, url);
+                    nf.error("Error adding magnet");
                 }
```

Comment:
I just checked and all notifications use double quotes. 
Reverting all to single quotes means that any word like "doesn't" will become a little bit of a pain, wouldn't it?